fink_science/ztf/superluminous/processor.py

```diff
@@ -0,0 +1,146 @@
+@profile
+def superluminous_score(
+    cjd: pd.Series,
+    cfid: pd.Series,
+    cmagpsf: pd.Series,
+    csigmapsf: pd.Series,
+    candidate: pd.Series,
+    is_transient: pd.Series,
+) -> pd.Series:
+    """High level spark wrapper for the superluminous classifier on ztf data
+
+    Parameters
+    ----------
+    cjd: Spark DataFrame Column
+        JD times (vectors of floats)
+    cfid: Spark DataFrame Column
+        Filter IDs (vectors of str)
+    cmagpsf, csigmapsf: Spark DataFrame Columns
+        Magnitude and magnitude error from photometry (vectors of floats)
+    candidate: Spark DataFrame Column
+        Additionnal info about the source. Contains distnr, the angular
+        distance to the nearest reference source.
+    is_transient: Spark DataFrame Column
+        Is the source likely a transient.
+
+    Returns
+    -------
+    np.array
+        Superluminous supernovae classification probability vector
+        Return 0 if not enough points were available for feature extraction
+
+    Examples
+    --------
+    >>> from fink_utils.spark.utils import concat_col
+    >>> from pyspark.sql import functions as F
+    >>> from fink_filters.ztf.filter_transient_complete.filter import transient_complete_filter
+    >>> from fink_science.ztf.transient_features.processor import extract_transient_features
+    >>> sdf = spark.read.load(ztf_alert_sample)
+    >>> sdf = extract_transient_features(sdf)
+    >>> sdf = sdf.withColumn(
+    ... "is_transient",
+    ... transient_complete_filter(
+    ... "faint", "positivesubtraction", "real", "pointunderneath",
+    ... "brightstar", "variablesource", "stationary", "roid"))
+
+    # Required alert columns
+    >>> what = ['jd', 'fid', 'magpsf', 'sigmapsf']
+
+    # Use for creating temp name
+    >>> prefix = 'c'
+    >>> what_prefix = [prefix + i for i in what]
+
+    # Append temp columns with historical + current measurements
+    >>> for colname in what:
+    ...     sdf = concat_col(sdf, colname, prefix=prefix)
+
+    # Perform the fit + classification (default model)
+    >>> args = [F.col(i) for i in what_prefix]
+    >>> args += ["candidate", "is_transient"]
+    >>> sdf = sdf.withColumn('proba', superluminous_score(*args))
+    >>> sdf.filter(sdf['proba']==-1).count()
+    55
+    >>> sdf.filter(sdf['proba']==0).count()
+    2
+    """
+    pdf = pd.DataFrame(
+        {
+            "cjd": cjd,
+            "cmagpsf": cmagpsf,
+            "csigmapsf": csigmapsf,
+            "cfid": cfid,
+            "distnr": candidate["distnr"],
+            "is_transient": is_transient,
+        }
+    )
+
+    # If no alert pass the transient filter,
+    # directly return invalid value for everyone.
+    if sum(pdf["is_transient"]) == 0:
+        return pd.Series([-1]*len(pdf))
+
+    else:
+        # Assign default 0 proba for every alert
+        probas = np.zeros(len(pdf))
+
+        # Assign -1 for non-transient alerts
+        probas[~pdf["is_transient"]] = -1
```

Comment:
Here you assign a probability of -1, but all the alert enters the module nevertheless (so you do not gain as much as you can in terms of performance). I propose in pseudo-code:

```python
if sum(pdf["is_transient"]) == 0:
    return pd.Series([-1.0]*len(pdf))
else:
  # Initialise all probas to -1
  probas_total = np.zeros(len(pdf), dtype=float) - 1
  
  # select only valid alerts
  mask_valid = pdf["is_transient"]
  pdf_valid = pdf[mask_valid]
  
  # then you do everything with `pdf_valid`
  
  probas_total[mask_valid] = probas_valid

return pd.Series(probas_total)
```

Do not hesitate to check indices before/after the change are correct (easy to mismatch array alignment!)